app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,30 @@
+class ApplicationsController < ApplicationController
+    def show
+        @application = Application.find(params[:id])
+        if params[:search].present?
+            @search_results = Pet.search(params[:search])
+        else
+            @search_results = []
+        end
+    end
+
+    def new
+        
+    end
+
+    def create
+        @application = Application.new(application_params)
+        if @application.save
+            redirect_to "/applications/#{@application.id}"
+        else
+            redirect_to "/applications/new"
+            flash[:alert] = "Error: #{error_message(@application.errors)}"
```

Comment:
Nice work making use of the inherited `error_message` method to format those errors! Great sad path handling